src/lib/parsers/sol-json.c

```diff
@@ -690,6 +690,24 @@ sol_json_serialize_boolean(struct sol_buffer *buffer, bool val)
     return 0;
 }
 
+static int
+_hex_decode(const char *p, uint8_t *val)
+{
+    int16_t n;
+    char hexdigits[3], *endptr;
+
+    hexdigits[2] = '\0';
+    memcpy(hexdigits, p, 2);
+    errno = 0;
+    n = strtoul(hexdigits, &endptr, 16);
```

Comment:
And strtoul returns negative numbers casted as unsined int. I am getting it as a signed int to check for negative values. But anyway I prefer your other solution